src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/Connection.cs

```diff
@@ -156,6 +157,21 @@ public virtual void OnSocketClosed()
             _socketClosedTcs.TrySetResult(null);
         }
 
+        // Called on Libuv thread
+        public void Tick()
+        {
+            if (_frame.Status == Frame.RequestProcessingStatus.RequestPending && // we're in between requests and
+                !SocketInput.IsCompleted)                                        // we haven't just started a new request
```

Comment:
This is meant to be checking the connection's SocketInput not the frame's right?